subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/cntr.c

```diff
@@ -11,21 +11,73 @@
 
 #include "hal/debug.h"
 
+#if !defined(CONFIG_BT_CTLR_NRF_GRTC)
 #ifndef NRF_RTC
 #if defined(CONFIG_SOC_COMPATIBLE_NRF54LX)
 #define NRF_RTC NRF_RTC10
 #else /* !CONFIG_SOC_COMPATIBLE_NRF54LX */
 #define NRF_RTC NRF_RTC0
 #endif /* !CONFIG_SOC_COMPATIBLE_NRF54LX */
 #endif /* !NRF_RTC */
+#endif /* CONFIG_BT_CTLR_NRF_GRTC */
 
 static uint8_t _refcount;
 
 void cntr_init(void)
 {
+#if defined(CONFIG_BT_CTLR_NRF_GRTC)
+#if defined(CONFIG_BT_CTLR_NRF_GRTC_START)
+	NRF_GRTC->MODE = (GRTC_MODE_SYSCOUNTEREN_Disabled <<
+			  GRTC_MODE_SYSCOUNTEREN_Pos) &
+			 GRTC_MODE_SYSCOUNTEREN_Msk;
+
+	NRF_GRTC->TASKS_CLEAR = 1U;
```

Comment:
You need to use the HAL for all tasks, clearing events, accessing CCL/H, CCEN and anything else w sideeffects.
As is now this just hangs in simulation.

Reply:
Do you have a list of NRF_GRTC registers with required sideeffects?

Reply:
https://github.com/BabbleSim/ext_nRF_hw_models/blob/main/src/HW_models/NHW_GRTC.h#L19-L40

Reply:
This is cool, perfect

Reply:
@aescolar correct me, i could not find bsim hal interfaces for dppi publish for grtc, ppib etc... I have reverted to SOC_SERIES_NRF54LX for NRF_GRTC use. Existing nrf54lxbsim will use NRF_RTC  instead.

Reply:
The files I pointed at contain the registers for which you need to ensure you use the nrf hal. For the others you can use the hal, or if you fancy just do direct register access.
In any case, you still have the *whole* nrf hal interface from the hal headers which will work fine in simulation.